_docs/install/kubernetes/index.md

```diff
@@ -57,3 +63,246 @@ $ helm delete --purge my-release
+
+    ```
+    kubectl create -f - <<END
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: storageos
+      labels:
+        app: storageos
+    END
+    ```
+
+1. Create role
+
+    ```
+    kubectl create -f - <<END
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: Role
+    metadata:
+      name: storageos
+      labels:
+        app: storageos
+    rules:
+    - apiGroups:
+        - ""
+      resources:
+        - secrets
+      verbs:
+        - create
+        - get
+        - list
+        - delete
+    END
+    ```
+
+1. Create role binding
+
+    ```
+    kubectl create -f - <<END
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: RoleBinding
+    metadata:
+      name: storageos
+      labels:
+        app: storageos
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: Role
+      name: storageos
+    subjects:
+      - kind: ServiceAccount
+        name: storageos
+    END
+    ```
+
+1. Create StorageOS service
+
+    ```
+    kubectl create -f - <<END
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: storageos
+      namespace: default
+      labels:
+        app: storageos
+    spec:
+      type: ClusterIP
+      ports:
+        - port: 5705
+          targetPort: 5705
+          protocol: TCP
+          name: storageos
+      selector:
+        app: storageos
+    END
+    ```
+
+
+1. Create secret
+
+    ```
+    CLUSTER_IP=$(kubectl get svc/storageos -o custom-columns=IP:spec.clusterIP --no-headers=true)
+    API_ADDRESS=$(echo -n "tcp://$CLUSTER_IP:5705" | base64)
+    kubectl create -f - <<END
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: storageos-api
+      namespace: default
+      labels:
+        app: storageos
+    type: "kubernetes.io/storageos"
+    data:
+      # Address build from the ClusterIP of the storageos service. In the format $(echo -n "tcp://$ClusterIP:5705" | base64)
+      apiAddress: "$API_ADDRESS"
+      apiUsername: "c3RvcmFnZW9z"
+      apiPassword: "c3RvcmFnZW9z"
+    END
+    ```
+
+    The secret hosts information regarding the API endpoint of StorageOS. It is required to add the StorageOS service IP in base64 for the `apiAddress` field.
+
+    `apiUsername` and `apiPassword` also need to be specified in base64. This example uses `echo "storageos" | base64` for both fields.
+
+1. Create storage class
+
+    ```
+    kubectl create -f - <<END
+    apiVersion: storage.k8s.io/v1beta1
+    kind: StorageClass
+    metadata:
+      name: fast
+      labels:
+        app: storageos
+    provisioner: kubernetes.io/storageos
+    parameters:
+      pool: default
+      description: Kubernetes volume
+      fsType: ext4
+      adminSecretNamespace: default
+      adminSecretName: storageos-api
+    END
+    ```
+
+    StorageOS uses a secret to discover the API endpoint of its workers. `adminSecretNamespace` have to be set according to the namespace created previously. Equally important is that
```

Comment:
StorageOS uses a secret to discover the API endpoint of its workers. `adminSecretNamespace` must be set to the specified namespace, and the service account needs to have granted permissions to StorageOS pods to have access to that secret.

(and same below)

Reply:
Done